### PR TITLE
RECORD: 1855 base + AWQ-lite mixed-precision GPTQ — val_bpb 1.06086 (3-seed mean)

### PR DESCRIPTION
**Pull request by aquariouseworkman**
Applies activation-aware mixed-precision GPTQ (from PR #1908 / romeerp) on top of codemath3000 PR #1855 stack.

## Results

| Seed | val_bpb (post-TTT) | artifact bytes | steps | eval time |
|------|--------------------|----------------|-------|-----------|
| 42   | 1.06118            | 15,978,503     | 4989  | 392.8s    |
| 314  | 1.06005            | 15,976,469     | 4986  | 395.8s    |
| 1234 | 1.06135            | 15,976,673     | 4977  | 395.5s    |
| **mean** | **1.06086**    | —              | —     | —         |

3-seed std: 0.00069. Beats codemath3000 PR #1855 (1.06108) by 0.00022 BPB.

## Technique

Training is identical to PR #1855. The only change is post-training quantization:

**AWQ-lite (activation-aware GPTQ):**
1. Collect per-input-channel activation RMS during GPTQ calibration
2. Score column groups: `saliency = act_rms * mean(abs(weight))`
3. Select top-1 most salient 64-column group per matrix
4. Quantize that group at int8 inside the same full-tensor GPTQ solve (rest stays int6)

Env vars: `AWQ_LITE_ENABLED=1 AWQ_LITE_BITS=8 AWQ_LITE_GROUP_TOP_K=1 AWQ_LITE_GROUP_SIZE=64`

## Setup
1. `pip install -r requirements.txt`
2. `apt-get install -y lrzip`
3. Install FA3: `pip install --no-deps flash_attn_3 --find-links https://windreamer.github.io/flash-attention3-wheels/cu128_torch291/`
4. Run `prepare_caseops_data.py` to build the dataset
5. `AWQ_LITE_ENABLED=1 AWQ_LITE_BITS=8 AWQ_LITE_GROUP_TOP_K=1 AWQ_LITE_GROUP_SIZE=64 torchrun --standalone --nproc_per_node=8 train_gpt.py`

## Environment
- 8xH100 80GB SXM (RunPod)
- PyTorch 2.9.1+cu128
- FlashAttention 3.0.0
- Triton 3.5.1

### CODE REVIEW

**Comment by h1beee** [comment]
> Beats codemath3000 PR (1.06108) by 0.00022 BPB

You need to outperform by 0.005 BPB: https://github.com/openai/parameter-golf#submission-process

**Comment by aquariouseworkman** [comment]
The code is byte-for-byte identical as https://github.com/openai/parameter-golf/pull/1908. I was just able to validate due to resources only, which Romeerp could not obtain. Due to his only limitation being the ability to acquire GPU resources and not ability to develop working code, the record obtained from this merge should go to Romeerp

**Comment by aquariouseworkman** [comment]
Closing, superseded by #1946, which is the clean cherry-picked AWQ-lite PR off openai:main. This branch (#1918 ) accidentally included the SmearGate BOS-fix commit, which already lives in #1851.

**Comment by akhoyannh-a11y** [comment]
Отправлено с iPhone29 апр. 2026 г., в 10:58, aquariouseworkman ***@***.***> написал(а):﻿Applies activation-aware mixed-precision GPTQ (from PR #1908 / romeerp) on top of codemath3000 PR #1855 stack.
Results



Seed
val_bpb (post-TTT)
artifact bytes
steps
eval time




42
1.06118
15,978,503
4989
392.8s


314
1.06005
15,976,469
4986
395.8s


1234
1.06135
15,976,673
4977
395.5s


mean
1.06086
—
—
—



3-seed std: 0.00069. Beats codemath3000 PR #1855 (1.06108) by 0.00022 BPB.
Technique
Training is identical to PR #1855. The only change is post-training quantization:
AWQ-lite (activation-aware GPTQ):

Collect per-input-channel activation RMS during GPTQ calibration
Score column groups: saliency = act_rms * mean(abs(weight))
Select top-1 most salient 64-column group per matrix
Quantize that group at int8 inside the same full-tensor GPTQ solve (rest stays int6)

Env vars: AWQ_LITE_ENABLED=1 AWQ_LITE_BITS=8 AWQ_LITE_GROUP_TOP_K=1 AWQ_LITE_GROUP_SIZE=64
Setup

pip install -r requirements.txt
apt-get install -y lrzip
Install FA3: pip install --no-deps flash_attn_3 --find-links https://windreamer.github.io/flash-attention3-wheels/cu128_torch291/
Run prepare_caseops_data.py to build the dataset
AWQ_LITE_ENABLED=1 AWQ_LITE_BITS=8 AWQ_LITE_GROUP_TOP_K=1 AWQ_LITE_GROUP_SIZE=64 torchrun --standalone --nproc_per_node=8 train_gpt.py

Environment

8xH100 80GB SXM (RunPod)
PyTorch 2.9.1+cu128
FlashAttention 3.0.0
Triton 3.5.1




You can view, comment on, or merge this pull request online at:
  https://github.com/openai/parameter-golf/pull/1918

Commit Summary

  58ee03b  Record: SmearGate BOS Fix + PR #1787 Base + Smear Gate + LQER Asymmetric + Phased TTT
  b346431  Record: base + AWQ-lite mixed-precision GPTQ — val_bpb 1.06086 (3-seed mean)
  d5328fe  Merge branch 'openai:main' into main


File Changes  (17 files)

  
    A
    records/track_10min_16mb/2020-04-29_AWQ_lite_mixedprecision_GPTQ/README.md
    (39)
  
  
    A
    records/track_10min_16mb/2020-04-29_AWQ_lite_mixedprecision_GPTQ/lossless_caps.py
    (833)
  
  
    A
    records/track_10min_16mb/2020-04-29_AWQ_lite_mixedprecision_GPTQ/prepare_caseops_data.py
    (177)
  
  
    A
    records/track_10min_16mb/2020-04-29_AWQ_lite_mixedprecision_GPTQ/requirements.txt
    (13)
  
  
    A
    records/track_10min_16mb/2020-04-29_AWQ_lite_mixedprecision_GPTQ/submission.json
    (25)
  
  
    A
    records/track_10min_16mb/2020-04-29_AWQ_lite_mixedprecision_GPTQ/tokenizers/fineweb_8192_bpe_lossless_caps_caseops_v1_reserved.model
    (0)
  
  
    A
    records/track_10min_16mb/2020-04-29_AWQ_lite_mixedprecision_GPTQ/train_gpt.py
    (3998)
  
  
    A
    records/track_10min_16mb/2020-04-29_AWQ_lite_mixedprecision_GPTQ/train_seed1234.log
    (947)
  
  
    A
    records/track_10min_16mb/2020-04-29_AWQ_lite_mixedprecision_GPTQ/train_seed314.log
    (951)
  
  
    A
    records/track_10min_16mb/2020-04-29_AWQ_lite_mixedprecision_GPTQ/train_seed42.log
    (950)
  
  
    A
    records/track_10min_16mb/2026-04-27_SmearGateBOSFix_PR1787Base_LQERAsym_PhasedTTT/README.md
    (54)
  
  
    A
    records/track_10min_16mb/2026-04-27_SmearGateBOSFix_PR1787Base_LQERAsym_PhasedTTT/lossless_caps.py
    (833)
  
  
    A
    records/track_10min_16mb/2026-04-27_SmearGateBOSFix_PR1787Base_LQERAsym_PhasedTTT/prepare_caseops_data.py
    (177)
  
  
    A
    records/track_10min_16mb/2026-04-27_SmearGateBOSFix_PR1787Base_LQERAsym_PhasedTTT/submission.json
    (30)
  
  
    A
    records/track_10min_16mb/2026-04-27_SmearGateBOSFix_PR1787Base_LQERAsym_PhasedTTT/tokenizers/fineweb_8192_bpe_lossless_caps_caseops_v1_reserved.model
    (0)
  
  
    A
    records/track_10min_16mb/2026-04-27_SmearGateBOSFix_PR1787Base_LQERAsym_PhasedTTT/train_gpt.py
    (3555)
  
  
    A
    records/track_10min_16mb/2026-04-27_SmearGateBOSFix_PR1787Base_LQERAsym_PhasedTTT/train_seed42.log
    (833)
  


Patch Links:

  https://github.com/openai/parameter-golf/pull/1918.patch
  https://github.com/openai/parameter-golf/pull/1918.diff
